src/pipecat/services/google.py

```diff
@@ -137,3 +144,195 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
+        logger.debug(f"Switching TTS pitch to: [{pitch}]")
+        self._params.pitch = pitch
+
+    async def set_rate(self, rate: str) -> None:
+        logger.debug(f"Switching TTS rate to: [{rate}]")
+        self._params.rate = rate
+
+    async def set_volume(self, volume: str) -> None:
+        logger.debug(f"Switching TTS volume to: [{volume}]")
+        self._params.volume = volume
+
+    async def set_emphasis(
+        self, emphasis: Literal["strong", "moderate", "reduced", "none"]
+    ) -> None:
+        logger.debug(f"Switching TTS emphasis to: [{emphasis}]")
+        self._params.emphasis = emphasis
+
+    async def set_gender(self, gender: Literal["male", "female", "neutral"]) -> None:
+        logger.debug(f"Switch TTS gender to [{gender}]")
+        self._params.gender = gender
+
+    async def google_style(
+        self, google_style: Literal["apologetic", "calm", "empathetic", "firm", "lively"]
+    ) -> None:
+        logger.debug(f"Switching TTS google style to: [{google_style}]")
+        self._params.google_style = google_style
+
+    async def set_params(self, params: InputParams) -> None:
+        logger.debug(f"Switching TTS params to: [{params}]")
+        self._params = params
+
+    async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
+        logger.debug(f"Generating TTS: [{text}]")
+
+        try:
+            await self.start_ttfb_metrics()
+
+            ssml = self._construct_ssml(text)
+            synthesis_input = texttospeech_v1.SynthesisInput(ssml=ssml)
+            voice = texttospeech_v1.VoiceSelectionParams(
+                language_code=self._params.language, name=self._voice_id
+            )
+            audio_config = texttospeech_v1.AudioConfig(
+                audio_encoding=texttospeech_v1.AudioEncoding.LINEAR16,
+                sample_rate_hertz=self.sample_rate,
+            )
+
+            request = texttospeech_v1.SynthesizeSpeechRequest(
+                input=synthesis_input, voice=voice, audio_config=audio_config
+            )
+
+            response = await self._client.synthesize_speech(request=request)
+
+            await self.start_tts_usage_metrics(text)
+
+            await self.push_frame(TTSStartedFrame())
+
+            # The audio produced by the TTS service has an audible click or pop at the beginning.
+            # This is due to the abrupt start of the audio waveform. To mitigate this, we apply a
+            # short fade-in effect to the audio data.
+
+            # Convert the response to a mutable numpy array
+            audio_content = np.frombuffer(response.audio_content, dtype=np.int16).copy()
+
+            # Apply a smooth, short fade-in
+            fade_duration = int(0.01 * self.sample_rate)  # 10ms fade-in
+            fade_in = np.square(
+                np.linspace(0, 1, fade_duration)
+            )  # Quadratic fade for smoother start
+            audio_content[:fade_duration] = audio_content[:fade_duration] * fade_in
```

Comment:
Maybe the reason for this is that they send WAV format. You can skip those bytes if that's the case. So, hopefully it's this and we can skip the fade-in part.

```
audio_content = audio_content[44:]
```

Reply:
That did the trick. Thank you!